File: backend/doc_configs/baa.py
from typing import Optional
from pydantic import BaseModel
from doc_configs.shared import PartialPartyInfo


SYSTEM_PROMPT = """You are a legal document assistant helping a user fill in a Business Associate Agreement (BAA) Cover Page.

## Fields you must collect

1. **BAA Effective Date** -- when the BAA takes effect
   Ask: "When should this BAA take effect? (e.g. 'today', 'April 1 2026', 'next Monday')"
   For shortcuts like "today", "tomorrow", "Mon", "next Mon", use this lookup table:
{date_lookup}
   For full dates (e.g. "March 15 2026", "2026-03-15"), convert to YYYY-MM-DD format directly.

2. **Breach Notification Period** -- time to report unauthorized use or disclosure of PHI
   Ask: "How quickly must breaches involving PHI be reported? (e.g. '5 business days', '72 hours', '10 calendar days')"

3. **Governing Law** -- jurisdiction whose laws govern the agreement
   Ask: "Which state or jurisdiction's law should govern this BAA? (e.g. 'California', 'New York', 'Texas')"

4. **Chosen Courts** -- which courts handle disputes
   Ask: "Which courts should have jurisdiction over disputes? (e.g. 'San Francisco, California', 'Houston, Texas')"

5. **Party 1 (Company)** -- name, title, company, notice address, signing date
   Ask for all sub-fields. Example: "Jane Smith, CEO of HealthCo, jane@healthco.com, signing today"

6. **Party 2 (Provider)** -- same sub-fields as Party 1

## Current document state
{current_fields}

## Instructions
- Work through the fields in order, but adapt naturally to what the user already told you
- Ask 1-2 questions per turn maximum
- Always include a concrete example in parentheses when asking a question
- When the user answers, confirm what you captured then always end with a follow-up question
- Once all fields are collected, tell the user the document is complete and they can download the PDF

## Clearing fields
If the user asks to clear fields (e.g. "clear all", "reset", "clear the date"):
- To clear a specific field, return that field set to its empty value and confirm what was cleared
- To clear all fields, return every field reset to empty values, confirm, then ask the first question again
- Never clear silently

Return JSON with:
- "message": your conversational reply (include examples when asking questions)
- "fields": only the fields you are confident about from this turn; omit everything else
"""


class PartialFields(BaseModel):
    baaEffectiveDate: Optional[str] = None
    breachNotificationPeriod: Optional[str] = None
    governingLaw: Optional[str] = None
    chosenCourts: Optional[str] = None
    party1: Optional[PartialPartyInfo] = None
    party2: Optional[PartialPartyInfo] = None


class AiResponse(BaseModel):
    message: str
    fields: PartialFields

File: frontend/lib/schemas/baa.ts
import { DocSchema } from "../doc-schema";

export const baaSchema: DocSchema = {
  id: "baa",
  title: "Business Associate Agreement",
  subtitle: "Cover Page",
  infoBox:
    'This Business Associate Agreement (the "BAA") consists of: (1) this Cover Page and (2) the Common Paper BAA Standard Terms Version 1.0 identical to those posted at commonpaper.com/standards/business-associate-agreement/1.0. The BAA governs the handling of protected health information (PHI) in compliance with HIPAA requirements.',
  coverFields: [
    {
      key: "baaEffectiveDate",
      label: "BAA Effective Date",
      type: "date",
      placeholder: "Select a date",
    },
    {
      key: "breachNotificationPeriod",
      label: "Breach Notification Period",
      hint: "Time to report unauthorized use or disclosure of PHI",
      type: "text",
      placeholder: "e.g. 5 business days",
    },
    {
      key: "governingLaw",
      label: "Governing Law",
      type: "text",
      placeholder: "State",
    },
    {
      key: "chosenCourts",
      label: "Chosen Courts",
      type: "text",
      placeholder: "City/county and state",
    },
  ],
  parties: [
    { label: "Company", fields: ["name", "title", "company", "noticeAddress", "date"] },
    { label: "Provider", fields: ["name", "title", "company", "noticeAddress", "date"] },
  ],
  signingNote: "By signing this Cover Page, each party agrees to enter into this Business Associate Agreement as of the BAA Effective Date.",
  templateFile: "templates/BAA.md",
  footerNote: "Common Paper Business Associate Agreement (Version 1.0) free to use under CC BY 4.0.",
  greeting:
    "Hi! I'm your BAA assistant. I'll help you fill in the Business Associate Agreement Cover Page through conversation. This governs how protected health information (PHI) is handled in compliance with HIPAA. Let's start -- when should this BAA take effect? (e.g. 'Today', 'April 1 2026', 'Immediately')",
  pdfFilename: "business-associate-agreement.pdf",
};
